samples/Quickstarts/2_InteractiveAspNetCore/src/IdentityServer/Quickstart/Device/DeviceController.cs

```diff
@@ -24,25 +26,30 @@ public class DeviceController : Controller
         private readonly IClientStore _clientStore;
         private readonly IResourceStore _resourceStore;
         private readonly IEventService _events;
+        private readonly IOptions<IdentityServerOptions> _options;
         private readonly ILogger<DeviceController> _logger;
 
         public DeviceController(
             IDeviceFlowInteractionService interaction,
             IClientStore clientStore,
             IResourceStore resourceStore,
             IEventService eventService,
+            IOptions<IdentityServerOptions> options,
             ILogger<DeviceController> logger)
         {
             _interaction = interaction;
             _clientStore = clientStore;
             _resourceStore = resourceStore;
             _events = eventService;
+            _options = options;
             _logger = logger;
         }
 
         [HttpGet]
-        public async Task<IActionResult> Index([FromQuery(Name = "user_code")] string userCode)
+        public async Task<IActionResult> Index()
         {
+            string userCodeParamName = _options.Value.UserInteraction.DeviceVerificationUserCodeParameter;
+            string userCode = Request.Query[userCodeParamName];
```

Comment:
Since `VerificationUriComplete` is generated based on this option, I think it's the only way to properly reflect the configuration